lib/chef/provider/package/apt.rb

```diff
@@ -89,11 +99,19 @@ def locked_packages
         end
 
         def install_package(name, version)
-          package_name = name.zip(version).map do |n, v|
-            package_data[n][:virtual] ? n : "#{n}=#{v}"
+          if source_files_exist?
+            sources = name.map { |n| name_sources[n] }
+            logger.info("#{new_resource} installing package(s): #{name.join(" ")}")
+            run_noninteractive("dpkg", "-i", *options, *sources)
```

Comment:
As @lamont-granquist [mentioned on the issue](https://github.com/chef/chef/issues/11103#issuecomment-809581119), we don't want to use dpkg in the apt provider. The notably difference is that dpkg won't attempt to install any dependencies and will fail if they're unmet, while apt will. If this was the dpkg provider that would be expected and okay, but this is the apt provider.

`apt install`:
```
$ sudo apt install ./apache2_2.4.41-4ubuntu3.8_amd64.deb
[sudo] password for btm:
Reading package lists... Done
Building dependency tree
Reading state information... Done
Note, selecting 'apache2' instead of './apache2_2.4.41-4ubuntu3.8_amd64.deb'
The following packages were automatically installed and are no longer required:
  libfprint-2-tod1 libllvm10:i386 libllvm9 libllvm9:i386 libnvidia-common-440 libnvidia-common-450 libnvidia-extra-440 libnvidia-extra-450 libnvidia-extra-460
  libsensors4-dev libwrap0-dev
Use 'sudo apt autoremove' to remove them.
The following additional packages will be installed:
  apache2-bin apache2-data apache2-utils libapr1 libaprutil1 libaprutil1-dbd-sqlite3 libaprutil1-ldap
Suggested packages:
  apache2-doc apache2-suexec-pristine | apache2-suexec-custom
The following NEW packages will be installed:
  apache2 apache2-bin apache2-data apache2-utils libapr1 libaprutil1 libaprutil1-dbd-sqlite3 libaprutil1-ldap
0 upgraded, 8 newly installed, 0 to remove and 302 not upgraded.
Need to get 1,715 kB of archives.
After this operation, 7,504 kB of additional disk space will be used.
Do you want to continue? [Y/n]
```

`dpkg -i`:
```
$ sudo dpkg -i ./apache2_2.4.41-4ubuntu3.8_amd64.deb
Selecting previously unselected package apache2.
(Reading database ... 482077 files and directories currently installed.)
Preparing to unpack .../apache2_2.4.41-4ubuntu3.8_amd64.deb ...
Unpacking apache2 (2.4.41-4ubuntu3.8) ...
dpkg: dependency problems prevent configuration of apache2:
 apache2 depends on apache2-bin (= 2.4.41-4ubuntu3.8); however:
  Package apache2-bin is not installed.
 apache2 depends on apache2-data (= 2.4.41-4ubuntu3.8); however:
  Package apache2-data is not installed.
 apache2 depends on apache2-utils (= 2.4.41-4ubuntu3.8); however:
  Package apache2-utils is not installed.

dpkg: error processing package apache2 (--install):
 dependency problems - leaving unconfigured
Processing triggers for systemd (245.4-4ubuntu3.11) ...
Processing triggers for ufw (0.36-6) ...
Processing triggers for man-db (2.9.1-1) ...
Errors were encountered while processing:
 apache2
```